js/tests/replybuilder_spec.js

```diff
@@ -0,0 +1,156 @@
+		var assertSameAddressList = function(l1, l2) {
+			var rawL1 = l1.map(function(a) {
+				return a.email;
+			});
+			var rawL2 = l2.map(function(a) {
+				return a.email;
+			});
+			rawL1.sort();
+			rawL2.sort();
+			expect(rawL1).toEqual(rawL2);
+		};
+
+		// b -> a to a -as b
+		it('handles a one-on-one reply', function() {
+			var a = createAddress('a@domain.tld');
+			var b = createAddress('b@domain.tld');
+			messageBody.set('from', [b]);
+			messageBody.set('to', [a]);
+			messageBody.set('cc', []);
+			setEmail(message, a);
+
+			var reply = ReplyBuilder.buildReply(message, messageBody);
+
+			assertSameAddressList(reply.from, [a]);
+			assertSameAddressList(reply.to, [b]);
+			assertSameAddressList(reply.cc, []);
+		});
+
+		it('handles simple group reply', function() {
+			var a = createAddress('a@domain.tld');
+			var b = createAddress('b@domain.tld');
+			var c = createAddress('c@domain.tld');
+			messageBody.set('from', [a]);
+			messageBody.set('to', [b, c]);
+			messageBody.set('cc', []);
+			setEmail(message, b);
+
+			var reply = ReplyBuilder.buildReply(message, messageBody);
+
+			assertSameAddressList(reply.from, [b]);
+			assertSameAddressList(reply.to, [a, c]);
+			assertSameAddressList(reply.cc, []);
+		});
+
+
+		it('handles group reply with CC', function() {
+			var a = createAddress('a@domain.tld');
+			var b = createAddress('b@domain.tld');
+			var c = createAddress('c@domain.tld');
+			var d = createAddress('d@domain.tld');
+			messageBody.set('from', [a]);
+			messageBody.set('to', [b, c]);
+			messageBody.set('cc', [d]);
+			setEmail(message, b);
+
+			var reply = ReplyBuilder.buildReply(message, messageBody);
+
+			assertSameAddressList(reply.from, [b]);
+			assertSameAddressList(reply.to, [a, c]);
+			assertSameAddressList(reply.cc, [d]);
+		});
+
+		it('handles group reply of CC address', function() {
+			var a = createAddress('a@domain.tld');
+			var b = createAddress('b@domain.tld');
+			var c = createAddress('c@domain.tld');
+			var d = createAddress('d@domain.tld');
+			messageBody.set('from', [a]);
+			messageBody.set('to', [b, c]);
+			messageBody.set('cc', [d]);
+			setEmail(message, d);
+
+			var reply = ReplyBuilder.buildReply(message, messageBody);
+
+			assertSameAddressList(reply.from, [d]);
+			assertSameAddressList(reply.to, [a, b, c]);
+			assertSameAddressList(reply.cc, []);
+		});
+
+		it('handles group reply of CC address with many CCs', function() {
+			var a = createAddress('a@domain.tld');
+			var b = createAddress('b@domain.tld');
+			var c = createAddress('c@domain.tld');
+			var d = createAddress('d@domain.tld');
+			var e = createAddress('e@domain.tld');
+			messageBody.set('from', [a]);
+			messageBody.set('to', [b, c]);
+			messageBody.set('cc', [d, e]);
+			setEmail(message, e);
+
+			var reply = ReplyBuilder.buildReply(message, messageBody);
+
+			assertSameAddressList(reply.from, [e]);
+			assertSameAddressList(reply.to, [a, b, c]);
+			assertSameAddressList(reply.cc, [d]);
+		});
```

Comment:
@jancborchardt if you could come up with a few more typical and thus important from, to, cc combinations we should have covered, please add them as comment and I'll add them.